### DISCUSSION TITLE
Kafka Vs Pulsar for Milvus

### DISCUSSION THREAD

**Discussion by navneetrajput06** [milvus-io/milvus#31502] [Q&A and General discussion]
Hi Team,
Can someone suggest which is the best approach for messaging queue between kafka and Pulsar. and also do we have any benchmarking test or toor to compare these tools.
Can also suggest which dataset and benchmarking we should use to compare these two.

**Comment by yhmo** [comment]
No much difference.
The deployment of Pulsar/Kafka and the resources will determine the capability of pulsar/kafka to process messages. Different users get different results.

You can use this benchmark tool to compare: https://github.com/zilliztech/VectorDBBench

**Reply by navneetrajput06** [reply]
We tried running this benchmark but it uses batch jobs which doesn't really interacting to kafka. can you suggest something which interacts with kafka or pulsar.

**Reply by yhmo** [reply]
kafka/pulsar mainly handles DML(insert/delete/upsert) operations and timetick messages.
You can use one client with python sdk to call insert() to do batch insert repeatedly/continually. For example, generate 50000 vectors(512 dim), write a loop to call insert(), every time insert the 50000 vectors. Compare which server is faster, milvus+pulsar or milvus+kafka.

No official benchmark, what you compare is only your two servers, choose the faster one.
